File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <set>
#include <vector>

using namespace std;

int getSum(vector<int>& elements, int start, int range) {
    int sum = 0;
    for (int i = 0; i < range; i++)
        sum += elements[(start + i) % elements.size()];
    return sum;
}

int solution(vector<int> elements) {
    set<int> s;
    for (int i = 0; i < elements.size(); i++) {
        for (int j = 0 ; j < elements.size(); j++)
            s.insert(getSum(elements, i, j));
    }
    return s.size();
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        Set<Integer> s = new HashSet<>();
        for (int i = 1; i <= elements.length; i++) {
            for (int j = 0; j < elements.length; j++) {
                int sum = 0;
                int idx = 0;
                while (idx < i) {
                    sum += elements[(j + idx) % elements.length];
                    idx++;
                }
                s.add(sum);
            }
        }
        return s.size();
    }
}